xapian-glib/tests/database.c

```diff
@@ -134,6 +134,73 @@ database_writable_flags_no_termlist (void)
   delete_database ("glass-db");
 }
 
+static void
+database_writable_allterms (void)
+{
+  GError *error = NULL;
+  XapianWritableDatabase *wdb =
+    xapian_writable_database_new_full ("glass-db",
+                                       XAPIAN_DATABASE_ACTION_CREATE,
+                                       XAPIAN_DATABASE_BACKEND_GLASS,
+                                       0,
+                                       &error);
+
+  g_assert_nonnull (wdb);
+
+  g_object_add_weak_pointer (G_OBJECT (wdb), (gpointer *) &wdb);
+
+  XapianDocument *doc = xapian_document_new ();
+
+  g_object_add_weak_pointer (G_OBJECT (doc), (gpointer *) &doc);
+
+  xapian_document_add_term (doc, "one");
+  xapian_document_add_term (doc, "two");
+
+  xapian_writable_database_add_document (wdb, doc, NULL, &error);
+  g_object_unref (wdb);
+
+  g_object_unref (doc);
+
+  XapianDatabase *db =
+    xapian_database_new_with_path ("glass-db", &error);
+
+  g_object_add_weak_pointer (G_OBJECT (db), (gpointer *) &db);
```

Comment:
I can't think of another reason than wanting to set the pointer to `NULL` when the instance dies (e.g. to verify that nothing leaks the object after the last unref), but nothing actually checks that in this test, so I'd prefer a simple unref (or `g_clear_object()`)